utils/municipality.ts

```diff
@@ -0,0 +1,339 @@
+  serializableMunicipality: ISerializableMunicipality
+): TaskEither<Error, Option<BlobService.BlobResult>> =>
+  CodiceCatastale.decode(serializableMunicipality.codiceCatastale).fold(
+    errs =>
+      fromLeft(
+        new Error(`Cannot decode CodiceCatastale| ${readableReport(errs)}`)
+      ),
+    codiceCatastaleValue =>
+      writeBlobFromJson(
+        blobService,
+        calculateMunicipalityPath(codiceCatastaleValue),
+        `${codiceCatastaleValue}.json` as NonEmptyString,
+        JSON.stringify(serializableMunicipality.municipality) as NonEmptyString
+      )
+  );
+
+export const exportAbolishedMunicipality = (
+  blobService: BlobService
+): TaskEither<Error, ReadonlyArray<Option<BlobService.BlobResult>>> =>
+  loadMunicipalityToCatastale(blobService)
+    .chain(municipalityToCatastale =>
+      loadAbolishedMunicipalities(blobService, municipalityToCatastale)
+    )
+    .chain(abolishedMunicipalities =>
+      array.sequence(taskEither)(
+        abolishedMunicipalities.map(municipality =>
+          serializeMunicipalityToJson(blobService, municipality)
+        )
+      )
+    );
+
+export const exportCurrentMunicipalities = (
+  blobService: BlobService
+): TaskEither<Error, ReadonlyArray<Option<BlobService.BlobResult>>> =>
+  getCsvFromURL(ITALIAN_MUNICIPALITIES_URL, {
+    encoding: "latin1"
+    // tslint:disable-next-line: no-any
+  } as any)
+    .chain(csvContent =>
+      parseCsv(csvContent, currentMiunicipalitiesParserOption)
+    )
+    .chain(result =>
+      array.sequence(taskEither)(
+        result.map(r =>
+          decodeMunicipality(r).fold(
+            e =>
+              fromLeft(
+                new Error(`Cannot decode current municipality| ${e.toString()}`)
+              ),
+            municipality =>
+              serializeMunicipalityToJson(blobService, {
+                codiceCatastale: r[19],
+                municipality
+              })
+          )
+        )
+      )
+    );
+
+export const exportForeignMunicipalities = (
+  blobService: BlobService
+): TaskEither<Error, ReadonlyArray<Option<BlobService.BlobResult>>> =>
+  getFileFromBlob(
+    blobService,
+    MUNICIPALITIES_CONTAINER_NAME,
+    FOREIGN_COUNTRIES_BLOB_NAME
+  )
+    .chain(maybeCsv =>
+      maybeCsv.foldL(
+        () =>
+          fromLeft<Error, StringMatrix>(
+            new Error("Elenco codici e denominazioni is empty")
+          ),
+        csv => parseCsv(csv, optionCsvParseForeignCountries)
+      )
+    )
+    .chain(result =>
+      array.sequence(taskEither)(
+        result
+          .filter(r => r[9] !== "")
+          .map(r =>
+            decodeForeignCountry(r).fold(
+              e =>
+                fromLeft(
+                  new Error(
+                    `Cannot decode foreign municipality| ${e.toString()}`
+                  )
+                ),
+              municipality =>
+                serializeMunicipalityToJson(blobService, {
+                  codiceCatastale: r[9],
```

Comment:
maybe an object with a map between indexex (integers) and column names (constants) can be useful here